src/radical/pilot/utils/db_utils.py

```diff
@@ -119,86 +100,7 @@ def get_session_docs(db, sid, cache=None, cachedir=None):
-#
-def get_session_slothist(db, sid, cache=None, cachedir=None):
-    """
-    For all pilots in the session, get the slot lists and slot histories. and
-    return as list of tuples like:
-
-         [pid   ,     [     [host,   slot]],     [      [slotstate, timestamp]]]
-    tuple(string, list(tuple(string, int )), list(tuple (string   , datetime )))
-    """
-
-    docs = get_session_docs(db, sid, cache, cachedir)
-
-    ret = dict()
-
-    for pilot_doc in docs['pilot']:
-
-        pid          = pilot_doc['uid']
-        slot_names   = list()
-        slot_infos   = dict()
-        slot_started = dict()
-
-        nodes   = pilot_doc['nodes']
-        n_cores = pilot_doc['cores_per_node']
-
-        if  not nodes:
-          # print "no nodes in pilot doc for %s" % pid
-            continue
-
-        for node in nodes:
-            for core in range(n_cores):
-                slot_name = "%s:%s" % (node, core)
-                slot_names.append (slot_name)
-                slot_infos  [slot_name] = list()
-                slot_started[slot_name] = sys.maxsize
-
-        for task_doc in docs['task']:
-            if task_doc['pilot'] == pilot_doc['uid']:
-
-                started  = None
-                finished = None
-                for event in sorted (task_doc['state_history'],
-                                     key=lambda x: x['timestamp']):
-                    if started:
-                        finished = event['timestamp']
-                        break
-                    if event['state'] == AGENT_EXECUTING:
-                        started = event['timestamp']
-
-                if not started or not finished:
-                  # print "no start/end for t %s - ignored" % task_doc['uid']
-                    continue
-
-                for slot_id in task_doc['opaque_slots']:
-                    if slot_id not in slot_infos:
-                      # print "slot %s for pilot %s unknown - ignored" \
-                      #     % (slot_id, pid)
-                        continue
-
-                    slot_infos[slot_id].append([started, finished])
-                    slot_started[slot_id] = min(started, slot_started[slot_id])
-
-        for slot_id in slot_infos:
-            slot_infos[slot_id].sort(key=lambda x: float(x[0]))
-
-        # we use the startup time to sort the slot names, as that gives a nicer
-        # representation when plotting.  That sorting should probably move to
-        # the plotting tools though... (FIXME)
-        slot_names.sort(key=lambda x: slot_started[x])
-
-        ret[pid] = dict()
-        ret[pid]['started']    = pilot_doc['started']
-        ret[pid]['finished']   = pilot_doc['finished']
-        ret[pid]['slots']      = slot_names
-        ret[pid]['slot_infos'] = slot_infos
-
-    return ret
-
-
-# ------------------------------------------------------------------------------
-def get_session_events(db, sid, cache=None, cachedir=None):
+def get_session_events(sid, cachedir=None):
```

Comment:
should we keep `db` as an input parameter and later provide it to `get_session_docs`?